eidr/schema.go

```diff
@@ -0,0 +1,116 @@
+// This file is part of the go-meta library.
+//
+// Copyright (C) 2017 JAAK MUSIC LTD
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// If you have any questions please contact yo@jaak.io
+
+package eidr
+
+import (
+	"github.com/meta-network/go-meta/migrate"
+)
+
+// migrations is a set of database migrations to run on a SQLite3 database
+// to prepare it to index EIDR objects
+var migrations = migrate.NewMigrations()
+
+func init() {
+	// migration 1 creates indexes for the following artist properties:
+	//
+	// * Foo - link to spec foo
+	//
+	migrations.Add(1, `
+CREATE TABLE baseobject (
+	doi_id TEXT NOT NULL,
+	structural_type TEXT NOT NULL,
+	referent_type TEXT NOT NULL,
+	status TEXT NOT NULL,
+	resource_name TEXT NOT NULL,
+	resource_name_lang TEXT NOT NULL,
+	resource_name_class TEXT,
+	approximate_length TEXT
+);
+
+CREATE INDEX baseobject_doi_id_idx ON baseobject(doi_id);
+
+CREATE TABLE org (
+	id TEXT,
+	idtype TEXT,
+	display_name TEXT NOT NULL,
+	role TEXT NOT NULL,
+	base_doi_id TEXT NOT NULL,
+	FOREIGN KEY(base_doi_id) REFERENCES baseobject(doi_id)
+);
+
+CREATE INDEX obj_idx ON org(id, idtype, base_doi_id);
```

Comment:
Why define a compound index on all three columns rather than an index for each? Will queries always include all three columns?

Reply:
I believe so, yes; the id have to be followed by id_type, and the objects are only of interest in their relation to the base object.